### DISCUSSION TITLE
Caprover Reverse Proxy Configuration

### DISCUSSION THREAD

**Discussion by kaaax0815** [coder/code-server#2690] [Q&A] [answered]
What Upstream Address should I use when trying to use Caprover Nginx Reverse Proxy? Because I cannot simple use caddy, because port is already in use. But I found I can't use localhost, because it would point to it own container.

Do you know what Upstream Address I should use? 
`bind-addr: 127.0.0.1:8053`

I tried `localhost`, `127.0.0.1` and `<server-ip>` and but they wont work

I have the Nginx Reverse Proxy Docker Container from Caprover, and Code-server is running baremetal on the host.

**Comment by code-asher** [comment]
For the NGINX container to be able to proxy to something else on the same host I think you'll have to set up a network or use docker-compose which automatically creates a private network between the services. https://stackoverflow.com/questions/38378635/nginx-as-reverse-proxy-on-another-container-how-to-configure-the-config-files

But I think this requires code-server to also be in a container.

You could also probably use `--net=host` on the NGINX container and then I think it'll be able to see other ports on the host.

Lastly, you should be able to use the `docker0` interface to access ports on the host system. https://stackoverflow.com/questions/31324981/how-to-access-host-port-from-docker-container

**Reply by kaaax0815** [reply]
Thanks for the help, I asked Caprover if their is a way to use the `--net=host`-Parameter.
Because I would need to reinstall e.g. pip  every time...

**Comment by kaaax0815** [comment, marked as answer]
Fixed it my setting `bind-addr:` to IP from docker0 and Port. Then Nginx could reach it
